packages/frontend-app-api/report.api.md

```diff
@@ -158,6 +161,10 @@ export type CreateSpecializedAppOptions = {
       | ExtensionFactoryMiddleware
       | ExtensionFactoryMiddleware[];
     pluginInfoResolver?: FrontendPluginInfoResolver;
+    router?: {
+      matchRoutes: RoutingContextType['matchRoutes'];
+      generatePath: RoutingContextType['generatePath'];
+    };
```

Comment:
Do we need to expose this as something configurable? I'm thinking it's best to keep this internal to make it easier to make changes

Reply:
Point is to abstract this away though so that we can switch out the implementation in app-api without impact to consumers. Thinking at most we'd have a flag for it in these options?